frontend/talentsearch/src/js/components/PageContainer.tsx

```diff
@@ -21,32 +21,42 @@ interface MenuLinkProps {
   href: string;
   text: string;
   title?: string;
+  as?: "a" | "button" | typeof Link;
   isActive?: (href: string, path: string) => boolean;
+  onClick?: () => void;
 }
 
 export const MenuLink: React.FC<MenuLinkProps> = ({
   href,
   text,
   title,
+  as = Link,
   isActive = exactMatch,
+  ...rest
 }) => {
   const location = useLocation();
+  const El = as;
   return (
-    <Link
+    <El
       href={href}
```

Comment:
That's neat! Injecting the html element using a prop. However, I don't think buttons can have a `href` attribute.. can they?

Reply:
You are very right, thanks for catching that 🎉 . I have fixed that along with not rendering empty titles with [remove href on menu buttons](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3009/commits/c20848fec708547251b52795b14554ea2b15cfbd)